### PR TITLE
F keepcoords ghi45

### PR DESCRIPTION
**Pull request by sarahLy9**
Fixes issue #45 allowing user to retain original input columns when converting coordinates to sfc or sfc to coordinates. We often need to include easting/northings as well as geometry. This functionality removes necessity to make copies of those columns before converting.

### CODE REVIEW

File: R/coords.R [hunk 1]
```diff
@@ -8,19 +8,22 @@
...
-                             activate = TRUE) {
+                             activate = TRUE,
+                             retain_orig = FALSE) {
   if (!is.data.frame(x)) ps_error("x must inherit from a data.frame")
   chk_vector(coords)
   check_values(coords, "")
   check_dim(coords, values = c(2L:3L))
-  check_names(x, coords)
+  chk::check_names(x, coords)
   chk_string(sfc_name)
```
Comment (joethorley) [inline R/coords.R]:
```suggestion
chk_flag(activate)
chk_flag(retain_orig)
```

File: R/coords.R [hunk 2]
```diff
@@ -77,9 +82,10 @@ ps_coords_to_sfc <- function(x, coords = c("X", "Y"),
...
 #' @param Y A string of the name of the Y coordinate.
 #' @param Z A string of the name of the Z coordinate.
+#' @param retain_orig A a flag indicating if the input coordinates should be retained as columns in the dataframe.
 #' @return The modified object with the sfc column removed
 #' @export
-ps_sfc_to_coords <- function(x, sfc_name = ps_active_sfc_name(x), X = "X", Y = "Y", Z = "Z") {
+ps_sfc_to_coords <- function(x, sfc_name = ps_active_sfc_name(x), X = "X", Y = "Y", Z = "Z", retain_orig = FALSE) {
   if (!is.data.frame(x)) ps_error("x must inherit from a data.frame")
   chk_string(sfc_name)
   chk_string(X)
```
Comment (joethorley) [inline R/coords.R:92]:
```suggestion
  chk_string(X)
  chk_flag(retain_orig)
```

**Review by joethorley — CHANGES_REQUESTED** [review]
I've made some suggested code changes (just adding chk_flag where missing).
Also could you add some tests?

**Comment by sarahLy9** [comment]
done, thanks!

File: R/coords.R [hunk 3]
```diff
@@ -8,24 +8,28 @@
...
                              sfc_name = "geometry",
-                             activate = TRUE) {
+                             activate = TRUE,
+                             retain_orig = FALSE) {
   if (!is.data.frame(x)) ps_error("x must inherit from a data.frame")
   chk_vector(coords)
   check_values(coords, "")
   check_dim(coords, values = c(2L:3L))
-  check_names(x, coords)
+  chk::check_names(x, coords)
```
Comment (aylapear) [inline R/coords.R]: This is inconsistent with the rest of the code in the function. Why are you doing the `chk::` for the one call? Since the full chk package is imported and to be consistent with the rest of the code in this function I would drop the `chk::`.

File: R/coords.R [hunk 4]
```diff
@@ -8,24 +8,28 @@
...
   chk_string(sfc_name)
+chk_flag(activate)
+chk_flag(retain_orig)
```
Comment (aylapear) [inline R/coords.R]: Spacing is wrong on these lines, please indent to match other code.

File: tests/testthat/test-coords.R [hunk 5]
```diff
@@ -11,3 +11,46 @@ test_that("ps_coords_to_sfc when missnig values", {
...
+
+test_that("ps_coords_to_sfc retain_orig flag keeps input columns when TRUE", {
+  x <- data.frame(X = c(1,1,10), Y = c(1,10,1))
+  x$Row <- 2:4
+
+  y <- ps_coords_to_sfc(x, crs = 28992, activate = FALSE, retain_orig = TRUE)
+
+  expect_identical(x$X, y$X)
+  expect_identical(x$Y, y$Y)
+
```
Comment (aylapear) [inline tests/testthat/test-coords.R]: You can remove these extra rows at the end

**Review by aylapear — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -16,10 +15,8 @@ data.
 
 ## Installation
```
Comment (joethorley) [inline README.md:17]: Why did you remove the code block on installation code?

File: R/add-z.R [hunk 7]
```diff
@@ -15,7 +15,7 @@ ps_sfc_add_z <- function(x, sfc_column = ps_active_sfc_name(x), z_column = "Elev
   x %<>% ps_activate_sfc(sfc_column)
   x %<>% cbind(sf::st_coordinates(x))
 
-  sfg <- do.call(list, purrr::map(1:nrow(x), function(y){
+  sfg <- do.call(list, purrr::map(seq_len(nrow(x)), function(y){
```
Comment (joethorley) [inline R/add-z.R:18]: Nice on the use of seq_len

Comment (joethorley) [reply R/add-z.R:18]: Although technically it should have been a separate PR to make review easier.

File: R/centroid.R [hunk 8]
```diff
@@ -26,8 +26,6 @@ ps_sfc_centroid1 <- function(x, sfc_name = ps_active_sfc_name(x), by = character
 
   x %<>% ps_activate_sfc(sfc_name)
 
-  crs <- sf::st_crs(x)
```
Comment (joethorley) [inline R/centroid.R:29]: Good catch.

**Review by joethorley — CHANGES_REQUESTED** [review]
Looks good - I really just have a few comments and a question

File: README.md (hunk 6 above)
Comment (sarahLy9) [reply README.md:17]: good question that was not intentional. looking at the commit history for README.Rmd it looks like the last commit that made changes to it was by Evan on 2023-10-23.

Comment (sarahLy9) [reply README.md:17]: great question that was not intentional. 

It looks correct in the README.Rmd file. When I preview the README.md file it is showing up ok. 
image

File: R/add-z.R (hunk 7 above)
Comment (sarahLy9) [reply R/add-z.R:18]: noted, thank you

File: README.md (hunk 6 above)
Comment (sarahLy9) [reply README.md:17]: I think that it is a just an updated syntax? those changes occured when I rendered the .md file from the Rmd file and it is still showing up as code block in the README section on the gh site.

**Review by aylapear — APPROVED** [review]
Looks good to me. Please ensure you have tested, checked, ran the documentation and rendered the readme before merging things in. 

Once merged in you will need to update the version and news file.

File: README.md [hunk 9]
```diff
@@ -16,10 +15,8 @@ data.
...
 
-``` r
-# install.packages("remotes")
-remotes::install_github("poissonconsulting/poisspatial")
-```
+    # install.packages("remotes")
```
Comment (joethorley) [inline README.md:18]: I'm wondering if our installations are doing something different because when I pull the code and knit the README.Rmd into the README.md file I get

```
## Installation

``` r
# install.packages("remotes")
remotes::install_github("poissonconsulting/poisspatial")
`` `
```

Note I had to add the space to `` ` to make it view correctly in this issue.

Can you confirm what you get when you knit the README.md file?

**Review by joethorley — APPROVED** [review]
Thanks @sarahLy9 - I just have the one question.

File: README.md (hunk 9 above)
Comment (joethorley) [reply README.md:18]: @sarahLy9 - can you resolve this before merging?

**Review by joethorley — CHANGES_REQUESTED** [review]
See comment on README.md file

File: README.md (hunk 9 above)
Comment (sarahLy9) [reply README.md:18]: @joethorley this is what I see when I preview the README.md file. I think our installations must be doing something different? I think I am using the most current versions of R and rmarkdown (2.28). Is there another package that would effect the rendering I can make sure is up to date?  
image

If I delete the README.md file and re-generate it this is what the md file looks like 
image

**Review by joethorley — APPROVED** [review]

**Comment by joethorley** [comment]
OK I've approved - please merge and then we can dig into why we are getting different renderings of our .md files!

**Comment by sarahLy9** [comment]
@aylapear does this mean I am not authorized to bump the version?
image
